.github/workflows/image-builder.yml

```
@@ -0,0 +1,181 @@
+on:
+  workflow_call:
+    inputs:
+      folder:
+        description: 'Either "workspaces" or "graders"'
+        required: true
+        type: string
+      image-prefix:
+        description: 'The prefix for the image name (e.g. prairielearn/workspace)'
+        required: true
+        type: string
+      base-images:
+        description: 'The base images to build'
+        required: false
+        type: string
+        default: '[]'
+      images:
+        description: 'The images to build'
+        required: true
+        type: string
+      exclude:
+        description: 'The combinations to exclude'
+        required: false
+        type: string
+        default: '[]'
+
+jobs:
+  build-base-images:
+    strategy:
+      max-parallel: 5
+      matrix:
+        image: ${{ fromJSON(inputs.base-images) }}
+        os: [ubuntu-24.04, ubuntu-24.04-arm]
+        exclude: ${{ fromJSON(inputs.exclude) }}
+    runs-on: ${{ matrix.os }}
+    env:
+      CAN_PUSH_IMAGE: ${{ secrets.DOCKERHUB_USERNAME != '' && github.ref_name == github.event.repository.default_branch }}
+    steps:
+      # Checkout the repository with full depth so we can check if the workspace was modified
```

Nit, "checkout" is a noun, "check out" is the verb.

```suggestion
      # Check out the repository with full depth so we can check if the workspace was modified
```

Ditto elsewhere.